pandas/io/sas/sas_xport.py

```diff
@@ -367,8 +367,8 @@ def _read_header(self):
             fl = field["field_length"]
             if field["ntype"] == "numeric" and ((fl < 2) or (fl > 8)):
                 self.close()
-                msg = "Floating field width {0} is not between 2 and 8."
-                raise TypeError(msg.format(fl))
+                msg = "f{Floating field width {fl} is not between 2 and 8.}"
```

Comment:
The ```f``` is inside the string :)

Reply:
How very sloppy of me... Good eye, it has been fixed now.